{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summary\n",
    "\n",
    "----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "mysqld._mysqld_process"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO - root - Initialized logger\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n",
      "2016-06-10 10:08:22.664922"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR - root - Initialized logger\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "%run imports.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO - root - Initialized logger\n",
      "ERROR - root - Initialized logger\n",
      "DEBUG - datapkg.daemons - 'basedir': /home/kimlab1/strokach/anaconda\n",
      "INFO - root - MySQL database already running...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n",
      "2016-06-10 20:00:18.889928\n"
     ]
    }
   ],
   "source": [
    "%run mysqld.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'load_data'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "NOTEBOOK_NAME = 'load_data'\n",
    "NOTEBOOK_NAME"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'mysql://root:@localhost:8301/dbnsfp?unix_socket=/tmp/strokach/dbnsfp/mysql.sock'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CONNECTION_STRING = mysqld.get_connection_string(SCHEMA_NAME)\n",
    "CONNECTION_STRING"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "db = datapkg.FileToMySQL(CONNECTION_STRING, NOTEBOOK_NAME, None, echo=False, db_engine='MyISAM')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extract"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dbNSFPv3.2a.zip\r\n"
     ]
    }
   ],
   "source": [
    "!ls ../downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!7z x -y -bd -bb0 '../downloads/dbNSFPv3.2a.zip' -o'$NOTEBOOK_NAME'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 96G\r\n",
      "-rw------- 1 strokach kimlab  78K Mar 31 00:50 dbNSFP3.2a.readme.txt\r\n",
      "-rw------- 1 strokach kimlab 9.6G Jun  9 03:26 dbNSFP3.2a_variant.chr1\r\n",
      "-rw------- 1 strokach kimlab 3.8G Jun  9 11:29 dbNSFP3.2a_variant.chr10\r\n",
      "-rw------- 1 strokach kimlab 5.7G Jun  9 12:10 dbNSFP3.2a_variant.chr11\r\n",
      "-rw------- 1 strokach kimlab 5.2G Jun  9 13:07 dbNSFP3.2a_variant.chr12\r\n",
      "-rw------- 1 strokach kimlab 1.7G Jun  9 13:55 dbNSFP3.2a_variant.chr13\r\n",
      "-rw------- 1 strokach kimlab 3.1G Jun  9 14:16 dbNSFP3.2a_variant.chr14\r\n",
      "-rw------- 1 strokach kimlab 3.3G Jun  9 17:25 dbNSFP3.2a_variant.chr15\r\n",
      "-rw------- 1 strokach kimlab 4.1G Jun  9 23:20 dbNSFP3.2a_variant.chr16\r\n",
      "-rw------- 1 strokach kimlab 5.5G Jun 10 02:07 dbNSFP3.2a_variant.chr17\r\n",
      "-rw------- 1 strokach kimlab 1.5G Jun 10 03:11 dbNSFP3.2a_variant.chr18\r\n",
      "-rw------- 1 strokach kimlab 6.1G Jun 10 03:32 dbNSFP3.2a_variant.chr19\r\n",
      "-rw------- 1 strokach kimlab 7.2G Jun  9 05:06 dbNSFP3.2a_variant.chr2\r\n",
      "-rw------- 1 strokach kimlab 2.3G Jun 10 04:39 dbNSFP3.2a_variant.chr20\r\n",
      "-rw------- 1 strokach kimlab 996M Jun 10 05:02 dbNSFP3.2a_variant.chr21\r\n",
      "-rw------- 1 strokach kimlab 2.0G Jun 10 05:14 dbNSFP3.2a_variant.chr22\r\n",
      "-rw------- 1 strokach kimlab 5.6G Jun  9 06:18 dbNSFP3.2a_variant.chr3\r\n",
      "-rw------- 1 strokach kimlab 3.8G Jun  9 07:13 dbNSFP3.2a_variant.chr4\r\n",
      "-rw------- 1 strokach kimlab 4.4G Jun  9 07:53 dbNSFP3.2a_variant.chr5\r\n",
      "-rw------- 1 strokach kimlab 4.9G Jun  9 08:39 dbNSFP3.2a_variant.chr6\r\n",
      "-rw------- 1 strokach kimlab 4.5G Jun  9 09:30 dbNSFP3.2a_variant.chr7\r\n",
      "-rw------- 1 strokach kimlab 3.2G Jun  9 10:16 dbNSFP3.2a_variant.chr8\r\n",
      "-rw------- 1 strokach kimlab 3.8G Jun  9 10:50 dbNSFP3.2a_variant.chr9\r\n",
      "-rw------- 1 strokach kimlab  17M Jun 10 05:33 dbNSFP3.2a_variant.chrM\r\n",
      "-rw------- 1 strokach kimlab 3.3G Jun 10 05:38 dbNSFP3.2a_variant.chrX\r\n",
      "-rw------- 1 strokach kimlab 177M Jun 10 06:13 dbNSFP3.2a_variant.chrY\r\n",
      "-rw------- 1 strokach kimlab 134M Jun  8 22:56 dbNSFP3.2_gene\r\n",
      "-rw------- 1 strokach kimlab 159M Jun  8 22:58 dbNSFP3.2_gene.complete\r\n",
      "-rw------- 1 strokach kimlab 3.2K May 31  2013 LICENSE.txt\r\n",
      "-rw------- 1 strokach kimlab  27K May  2 19:33 search_dbNSFP32a.class\r\n",
      "-rw------- 1 strokach kimlab  40K May  2 19:33 search_dbNSFP32a.java\r\n",
      "-rw------- 1 strokach kimlab  97K Mar 19 21:18 search_dbNSFP32a.readme.pdf\r\n",
      "-rw------- 1 strokach kimlab  258 Apr  1  2015 tryhg18.in\r\n",
      "-rw------- 1 strokach kimlab  256 Apr 12  2015 tryhg19.in\r\n",
      "-rw------- 1 strokach kimlab  254 Apr 12  2015 tryhg38.in\r\n",
      "-rw------- 1 strokach kimlab 5.9K Aug 12  2013 try.vcf\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh $NOTEBOOK_NAME"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load to database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene = pd.read_csv(\n",
    "    op.join(NOTEBOOK_NAME, 'dbNSFP3.2_gene'), sep='\\t', nrows=100, na_values=['.', '\\\\N'])\n",
    "gene.columns = datapkg.format_columns(gene.columns)\n",
    "display(gene.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "t = db.import_table(\n",
    "    filename=op.abspath(op.join(NOTEBOOK_NAME, 'dbNSFP3.2_gene')),\n",
    "    tablename='gene',\n",
    "    na_values=['.', '\\\\N'],\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "db.create_indexes(\n",
    "    t.name,\n",
    "    index_commands=[\n",
    "        [('gene_name', ), True], \n",
    "        [('ensembl_gene', ), False],\n",
    "        [('uniprot_acc', ), False],\n",
    "        [('uniprot_id', ), False],\n",
    "        [('entrez_gene_id', ), True],\n",
    "        [('mim_id', ), True],\n",
    "    ],\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## gene_complete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_complete = pd.read_csv(\n",
    "    op.join(NOTEBOOK_NAME, 'dbNSFP3.2_gene.complete'), sep='\\t', nrows=100, na_values=['.', '\\\\N'])\n",
    "gene_complete.columns = datapkg.format_columns(gene_complete.columns)\n",
    "display(gene_complete.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "t = db.import_table(\n",
    "    filename=op.abspath(op.join(NOTEBOOK_NAME, 'dbNSFP3.2_gene.complete')),\n",
    "    tablename='gene_complete',\n",
    "    na_values=['.', '\\\\N']\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "db.create_indexes(\n",
    "    t.name,\n",
    "    index_commands=[\n",
    "        [('gene_name', ), True], \n",
    "        [('ensembl_gene', ), False],\n",
    "        [('uniprot_acc', ), False],\n",
    "        [('uniprot_id', ), False],\n",
    "        [('entrez_gene_id', ), True],\n",
    "        [('mim_id', ), True],\n",
    "    ],\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Try to figure out why there are one -> many, uniprot_acc -> gene_name mappings\n",
    "sql_query = \"\"\"\\\n",
    "select uniprot_acc, group_concat(distinct gene_name) gene_names\n",
    "from gene_complete\n",
    "group by uniprot_acc\n",
    "having count(gene_name) > 1\n",
    "\"\"\"\n",
    "df = pd.read_sql_query(sql_query, db.engine)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "display(df.head())\n",
    "print(df.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chr</th>\n",
       "      <th>pos(1-based)</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>aaref</th>\n",
       "      <th>aaalt</th>\n",
       "      <th>rs_dbSNP146</th>\n",
       "      <th>hg19_chr</th>\n",
       "      <th>hg19_pos(1-based)</th>\n",
       "      <th>hg18_chr</th>\n",
       "      <th>hg18_pos(1-based)</th>\n",
       "      <th>genename</th>\n",
       "      <th>cds_strand</th>\n",
       "      <th>refcodon</th>\n",
       "      <th>codonpos</th>\n",
       "      <th>codon_degeneracy</th>\n",
       "      <th>Ancestral_allele</th>\n",
       "      <th>AltaiNeandertal</th>\n",
       "      <th>Denisova</th>\n",
       "      <th>Ensembl_geneid</th>\n",
       "      <th>Ensembl_transcriptid</th>\n",
       "      <th>Ensembl_proteinid</th>\n",
       "      <th>aapos</th>\n",
       "      <th>SIFT_score</th>\n",
       "      <th>SIFT_converted_rankscore</th>\n",
       "      <th>SIFT_pred</th>\n",
       "      <th>Uniprot_acc_Polyphen2</th>\n",
       "      <th>Uniprot_id_Polyphen2</th>\n",
       "      <th>Uniprot_aapos_Polyphen2</th>\n",
       "      <th>Polyphen2_HDIV_score</th>\n",
       "      <th>Polyphen2_HDIV_rankscore</th>\n",
       "      <th>Polyphen2_HDIV_pred</th>\n",
       "      <th>Polyphen2_HVAR_score</th>\n",
       "      <th>Polyphen2_HVAR_rankscore</th>\n",
       "      <th>Polyphen2_HVAR_pred</th>\n",
       "      <th>LRT_score</th>\n",
       "      <th>LRT_converted_rankscore</th>\n",
       "      <th>LRT_pred</th>\n",
       "      <th>LRT_Omega</th>\n",
       "      <th>MutationTaster_score</th>\n",
       "      <th>MutationTaster_converted_rankscore</th>\n",
       "      <th>MutationTaster_pred</th>\n",
       "      <th>MutationTaster_model</th>\n",
       "      <th>MutationTaster_AAE</th>\n",
       "      <th>MutationAssessor_UniprotID</th>\n",
       "      <th>MutationAssessor_variant</th>\n",
       "      <th>MutationAssessor_score</th>\n",
       "      <th>MutationAssessor_score_rankscore</th>\n",
       "      <th>MutationAssessor_pred</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>FATHMM_converted_rankscore</th>\n",
       "      <th>FATHMM_pred</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>PROVEAN_converted_rankscore</th>\n",
       "      <th>PROVEAN_pred</th>\n",
       "      <th>Transcript_id_VEST3</th>\n",
       "      <th>Transcript_var_VEST3</th>\n",
       "      <th>VEST3_score</th>\n",
       "      <th>VEST3_rankscore</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaSVM_rankscore</th>\n",
       "      <th>MetaSVM_pred</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>MetaLR_rankscore</th>\n",
       "      <th>MetaLR_pred</th>\n",
       "      <th>Reliability_index</th>\n",
       "      <th>CADD_raw</th>\n",
       "      <th>CADD_raw_rankscore</th>\n",
       "      <th>CADD_phred</th>\n",
       "      <th>DANN_score</th>\n",
       "      <th>DANN_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_score</th>\n",
       "      <th>fathmm-MKL_coding_rankscore</th>\n",
       "      <th>fathmm-MKL_coding_pred</th>\n",
       "      <th>fathmm-MKL_coding_group</th>\n",
       "      <th>Eigen-raw</th>\n",
       "      <th>Eigen-phred</th>\n",
       "      <th>Eigen-raw_rankscore</th>\n",
       "      <th>Eigen-PC-raw</th>\n",
       "      <th>Eigen-PC-raw_rankscore</th>\n",
       "      <th>GenoCanyon_score</th>\n",
       "      <th>GenoCanyon_score_rankscore</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>integrated_fitCons_score_rankscore</th>\n",
       "      <th>integrated_confidence_value</th>\n",
       "      <th>GM12878_fitCons_score</th>\n",
       "      <th>GM12878_fitCons_score_rankscore</th>\n",
       "      <th>GM12878_confidence_value</th>\n",
       "      <th>H1-hESC_fitCons_score</th>\n",
       "      <th>H1-hESC_fitCons_score_rankscore</th>\n",
       "      <th>H1-hESC_confidence_value</th>\n",
       "      <th>HUVEC_fitCons_score</th>\n",
       "      <th>HUVEC_fitCons_score_rankscore</th>\n",
       "      <th>HUVEC_confidence_value</th>\n",
       "      <th>GERP++_NR</th>\n",
       "      <th>GERP++_RS</th>\n",
       "      <th>GERP++_RS_rankscore</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>phyloP100way_vertebrate_rankscore</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phyloP20way_mammalian_rankscore</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>phastCons20way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_pi</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>1000Gp3_AC</th>\n",
       "      <th>1000Gp3_AF</th>\n",
       "      <th>1000Gp3_AFR_AC</th>\n",
       "      <th>1000Gp3_AFR_AF</th>\n",
       "      <th>1000Gp3_EUR_AC</th>\n",
       "      <th>1000Gp3_EUR_AF</th>\n",
       "      <th>1000Gp3_AMR_AC</th>\n",
       "      <th>1000Gp3_AMR_AF</th>\n",
       "      <th>1000Gp3_EAS_AC</th>\n",
       "      <th>1000Gp3_EAS_AF</th>\n",
       "      <th>1000Gp3_SAS_AC</th>\n",
       "      <th>1000Gp3_SAS_AF</th>\n",
       "      <th>TWINSUK_AC</th>\n",
       "      <th>TWINSUK_AF</th>\n",
       "      <th>ALSPAC_AC</th>\n",
       "      <th>ALSPAC_AF</th>\n",
       "      <th>ESP6500_AA_AC</th>\n",
       "      <th>ESP6500_AA_AF</th>\n",
       "      <th>ESP6500_EA_AC</th>\n",
       "      <th>ESP6500_EA_AF</th>\n",
       "      <th>ExAC_AC</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_Adj_AC</th>\n",
       "      <th>ExAC_Adj_AF</th>\n",
       "      <th>ExAC_AFR_AC</th>\n",
       "      <th>ExAC_AFR_AF</th>\n",
       "      <th>ExAC_AMR_AC</th>\n",
       "      <th>ExAC_AMR_AF</th>\n",
       "      <th>ExAC_EAS_AC</th>\n",
       "      <th>ExAC_EAS_AF</th>\n",
       "      <th>ExAC_FIN_AC</th>\n",
       "      <th>ExAC_FIN_AF</th>\n",
       "      <th>ExAC_NFE_AC</th>\n",
       "      <th>ExAC_NFE_AF</th>\n",
       "      <th>ExAC_SAS_AC</th>\n",
       "      <th>ExAC_SAS_AF</th>\n",
       "      <th>ExAC_nonTCGA_AC</th>\n",
       "      <th>ExAC_nonTCGA_AF</th>\n",
       "      <th>ExAC_nonTCGA_Adj_AC</th>\n",
       "      <th>ExAC_nonTCGA_Adj_AF</th>\n",
       "      <th>ExAC_nonTCGA_AFR_AC</th>\n",
       "      <th>ExAC_nonTCGA_AFR_AF</th>\n",
       "      <th>ExAC_nonTCGA_AMR_AC</th>\n",
       "      <th>ExAC_nonTCGA_AMR_AF</th>\n",
       "      <th>ExAC_nonTCGA_EAS_AC</th>\n",
       "      <th>ExAC_nonTCGA_EAS_AF</th>\n",
       "      <th>ExAC_nonTCGA_FIN_AC</th>\n",
       "      <th>ExAC_nonTCGA_FIN_AF</th>\n",
       "      <th>ExAC_nonTCGA_NFE_AC</th>\n",
       "      <th>ExAC_nonTCGA_NFE_AF</th>\n",
       "      <th>ExAC_nonTCGA_SAS_AC</th>\n",
       "      <th>ExAC_nonTCGA_SAS_AF</th>\n",
       "      <th>ExAC_nonpsych_AC</th>\n",
       "      <th>ExAC_nonpsych_AF</th>\n",
       "      <th>ExAC_nonpsych_Adj_AC</th>\n",
       "      <th>ExAC_nonpsych_Adj_AF</th>\n",
       "      <th>ExAC_nonpsych_AFR_AC</th>\n",
       "      <th>ExAC_nonpsych_AFR_AF</th>\n",
       "      <th>ExAC_nonpsych_AMR_AC</th>\n",
       "      <th>ExAC_nonpsych_AMR_AF</th>\n",
       "      <th>ExAC_nonpsych_EAS_AC</th>\n",
       "      <th>ExAC_nonpsych_EAS_AF</th>\n",
       "      <th>ExAC_nonpsych_FIN_AC</th>\n",
       "      <th>ExAC_nonpsych_FIN_AF</th>\n",
       "      <th>ExAC_nonpsych_NFE_AC</th>\n",
       "      <th>ExAC_nonpsych_NFE_AF</th>\n",
       "      <th>ExAC_nonpsych_SAS_AC</th>\n",
       "      <th>ExAC_nonpsych_SAS_AF</th>\n",
       "      <th>clinvar_rs</th>\n",
       "      <th>clinvar_clnsig</th>\n",
       "      <th>clinvar_trait</th>\n",
       "      <th>clinvar_golden_stars</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>GTEx_V6_gene</th>\n",
       "      <th>GTEx_V6_tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>M</td>\n",
       "      <td>L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>1</td>\n",
       "      <td>58954</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>ATG</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENST00000335137</td>\n",
       "      <td>ENSP00000334393</td>\n",
       "      <td>1</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.26702</td>\n",
       "      <td>T</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02634</td>\n",
       "      <td>B</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01257</td>\n",
       "      <td>B</td>\n",
       "      <td>0.589091</td>\n",
       "      <td>0.05577</td>\n",
       "      <td>N</td>\n",
       "      <td>1.33974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>M1L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.76</td>\n",
       "      <td>0.00529</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.38</td>\n",
       "      <td>0.13435</td>\n",
       "      <td>N</td>\n",
       "      <td>NM_001005484.1</td>\n",
       "      <td>M1L</td>\n",
       "      <td>0.120</td>\n",
       "      <td>0.13350</td>\n",
       "      <td>-0.9577</td>\n",
       "      <td>0.39629</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>T</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-1.436194</td>\n",
       "      <td>0.01840</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.443783</td>\n",
       "      <td>0.03370</td>\n",
       "      <td>0.02063</td>\n",
       "      <td>0.06083</td>\n",
       "      <td>N</td>\n",
       "      <td>AEFI</td>\n",
       "      <td>-1.394131</td>\n",
       "      <td>0.119256</td>\n",
       "      <td>0.02697</td>\n",
       "      <td>-1.535705</td>\n",
       "      <td>0.02036</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.0355</td>\n",
       "      <td>0.487112</td>\n",
       "      <td>0.13308</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.26071</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.22998</td>\n",
       "      <td>0</td>\n",
       "      <td>0.564101</td>\n",
       "      <td>0.26208</td>\n",
       "      <td>0</td>\n",
       "      <td>2.31</td>\n",
       "      <td>-4.63</td>\n",
       "      <td>0.03101</td>\n",
       "      <td>-0.055</td>\n",
       "      <td>0.11668</td>\n",
       "      <td>-1.983</td>\n",
       "      <td>0.00506</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.06329</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01567</td>\n",
       "      <td>0.2547:0.0:0.5282:0.2171</td>\n",
       "      <td>3.5592</td>\n",
       "      <td>0.07372</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>M</td>\n",
       "      <td>V</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>1</td>\n",
       "      <td>58954</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>ATG</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENST00000335137</td>\n",
       "      <td>ENSP00000334393</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00963</td>\n",
       "      <td>T</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02634</td>\n",
       "      <td>B</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01257</td>\n",
       "      <td>B</td>\n",
       "      <td>0.589091</td>\n",
       "      <td>0.05577</td>\n",
       "      <td>N</td>\n",
       "      <td>1.33974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>M1V</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.76</td>\n",
       "      <td>0.00529</td>\n",
       "      <td>T</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>N</td>\n",
       "      <td>NM_001005484.1</td>\n",
       "      <td>M1V</td>\n",
       "      <td>0.059</td>\n",
       "      <td>0.04258</td>\n",
       "      <td>-0.9620</td>\n",
       "      <td>0.38819</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>0.00188</td>\n",
       "      <td>T</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-3.657273</td>\n",
       "      <td>0.00025</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.263843</td>\n",
       "      <td>0.01262</td>\n",
       "      <td>0.00266</td>\n",
       "      <td>0.01460</td>\n",
       "      <td>N</td>\n",
       "      <td>AEFI</td>\n",
       "      <td>-1.588121</td>\n",
       "      <td>0.057701</td>\n",
       "      <td>0.01327</td>\n",
       "      <td>-1.701456</td>\n",
       "      <td>0.01108</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.0355</td>\n",
       "      <td>0.487112</td>\n",
       "      <td>0.13308</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.26071</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.22998</td>\n",
       "      <td>0</td>\n",
       "      <td>0.564101</td>\n",
       "      <td>0.26208</td>\n",
       "      <td>0</td>\n",
       "      <td>2.31</td>\n",
       "      <td>-4.63</td>\n",
       "      <td>0.03101</td>\n",
       "      <td>-0.055</td>\n",
       "      <td>0.11668</td>\n",
       "      <td>-1.983</td>\n",
       "      <td>0.00506</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.06329</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01567</td>\n",
       "      <td>0.2547:0.0:0.5282:0.2171</td>\n",
       "      <td>3.5592</td>\n",
       "      <td>0.07372</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #chr  pos(1-based) ref alt aaref aaalt rs_dbSNP146  hg19_chr  hg19_pos(1-based)  hg18_chr  hg18_pos(1-based) genename cds_strand refcodon  codonpos  codon_degeneracy  Ancestral_allele AltaiNeandertal Denisova   Ensembl_geneid Ensembl_transcriptid Ensembl_proteinid  aapos  SIFT_score  SIFT_converted_rankscore SIFT_pred Uniprot_acc_Polyphen2 Uniprot_id_Polyphen2  Uniprot_aapos_Polyphen2  Polyphen2_HDIV_score  Polyphen2_HDIV_rankscore Polyphen2_HDIV_pred  Polyphen2_HVAR_score  Polyphen2_HVAR_rankscore Polyphen2_HVAR_pred  LRT_score  LRT_converted_rankscore LRT_pred  LRT_Omega  MutationTaster_score  MutationTaster_converted_rankscore MutationTaster_pred MutationTaster_model MutationTaster_AAE MutationAssessor_UniprotID MutationAssessor_variant  MutationAssessor_score  MutationAssessor_score_rankscore MutationAssessor_pred  FATHMM_score  FATHMM_converted_rankscore FATHMM_pred  PROVEAN_score  PROVEAN_converted_rankscore PROVEAN_pred Transcript_id_VEST3 Transcript_var_VEST3  VEST3_score  \\\n",
       "0     1         69091   A   C     M     L         NaN         1              69091         1              58954    OR4F5          +      ATG         1                 0               NaN             ./.      ./.  ENSG00000186092      ENST00000335137   ENSP00000334393      1        0.13                   0.26702         T                Q8NH21          OR4F5_HUMAN                      1.0                   0.0                   0.02634                   B                   0.0                   0.01257                   B   0.589091                  0.05577        N    1.33974                   NaN                                 NaN                 NaN                  NaN                NaN                OR4F5_HUMAN                      M1L                     NaN                               NaN                   NaN          6.76                     0.00529           T          -0.38                      0.13435            N      NM_001005484.1                  M1L        0.120   \n",
       "1     1         69091   A   G     M     V         NaN         1              69091         1              58954    OR4F5          +      ATG         1                 0               NaN             ./.      ./.  ENSG00000186092      ENST00000335137   ENSP00000334393      1        1.00                   0.00963         T                Q8NH21          OR4F5_HUMAN                      1.0                   0.0                   0.02634                   B                   0.0                   0.01257                   B   0.589091                  0.05577        N    1.33974                   NaN                                 NaN                 NaN                  NaN                NaN                OR4F5_HUMAN                      M1V                     NaN                               NaN                   NaN          6.76                     0.00529           T           0.71                      0.02137            N      NM_001005484.1                  M1V        0.059   \n",
       "\n",
       "   VEST3_rankscore  MetaSVM_score  MetaSVM_rankscore MetaSVM_pred  MetaLR_score  MetaLR_rankscore MetaLR_pred  Reliability_index  CADD_raw  CADD_raw_rankscore  CADD_phred  DANN_score  DANN_rankscore  fathmm-MKL_coding_score  fathmm-MKL_coding_rankscore fathmm-MKL_coding_pred fathmm-MKL_coding_group  Eigen-raw  Eigen-phred  Eigen-raw_rankscore  Eigen-PC-raw  Eigen-PC-raw_rankscore  GenoCanyon_score  GenoCanyon_score_rankscore  integrated_fitCons_score  integrated_fitCons_score_rankscore  integrated_confidence_value  GM12878_fitCons_score  GM12878_fitCons_score_rankscore  GM12878_confidence_value  H1-hESC_fitCons_score  H1-hESC_fitCons_score_rankscore  H1-hESC_confidence_value  HUVEC_fitCons_score  HUVEC_fitCons_score_rankscore  HUVEC_confidence_value  GERP++_NR  GERP++_RS  GERP++_RS_rankscore  phyloP100way_vertebrate  phyloP100way_vertebrate_rankscore  phyloP20way_mammalian  phyloP20way_mammalian_rankscore  phastCons100way_vertebrate  phastCons100way_vertebrate_rankscore  \\\n",
       "0          0.13350        -0.9577            0.39629            T        0.0009           0.00318           T                8.0 -1.436194             0.01840       0.003    0.443783         0.03370                  0.02063                      0.06083                      N                    AEFI  -1.394131     0.119256              0.02697     -1.535705                 0.02036          0.000022                      0.0355                  0.487112                             0.13308                            0               0.573888                          0.26071                         0               0.573888                          0.22998                         0             0.564101                        0.26208                       0       2.31      -4.63              0.03101                   -0.055                            0.11668                 -1.983                          0.00506                         0.0                               0.06329   \n",
       "1          0.04258        -0.9620            0.38819            T        0.0005           0.00188           T                8.0 -3.657273             0.00025       0.001    0.263843         0.01262                  0.00266                      0.01460                      N                    AEFI  -1.588121     0.057701              0.01327     -1.701456                 0.01108          0.000022                      0.0355                  0.487112                             0.13308                            0               0.573888                          0.26071                         0               0.573888                          0.22998                         0             0.564101                        0.26208                       0       2.31      -4.63              0.03101                   -0.055                            0.11668                 -1.983                          0.00506                         0.0                               0.06329   \n",
       "\n",
       "   phastCons20way_mammalian  phastCons20way_mammalian_rankscore            SiPhy_29way_pi  SiPhy_29way_logOdds  SiPhy_29way_logOdds_rankscore  1000Gp3_AC  1000Gp3_AF  1000Gp3_AFR_AC  1000Gp3_AFR_AF  1000Gp3_EUR_AC  1000Gp3_EUR_AF  1000Gp3_AMR_AC  1000Gp3_AMR_AF  1000Gp3_EAS_AC  1000Gp3_EAS_AF  1000Gp3_SAS_AC  1000Gp3_SAS_AF  TWINSUK_AC  TWINSUK_AF  ALSPAC_AC  ALSPAC_AF  ESP6500_AA_AC  ESP6500_AA_AF  ESP6500_EA_AC  ESP6500_EA_AF  ExAC_AC  ExAC_AF  ExAC_Adj_AC  ExAC_Adj_AF  ExAC_AFR_AC  ExAC_AFR_AF  ExAC_AMR_AC  ExAC_AMR_AF  ExAC_EAS_AC  ExAC_EAS_AF  ExAC_FIN_AC  ExAC_FIN_AF  ExAC_NFE_AC  ExAC_NFE_AF  ExAC_SAS_AC  ExAC_SAS_AF  ExAC_nonTCGA_AC  ExAC_nonTCGA_AF  ExAC_nonTCGA_Adj_AC  ExAC_nonTCGA_Adj_AF  ExAC_nonTCGA_AFR_AC  ExAC_nonTCGA_AFR_AF  ExAC_nonTCGA_AMR_AC  ExAC_nonTCGA_AMR_AF  ExAC_nonTCGA_EAS_AC  ExAC_nonTCGA_EAS_AF  ExAC_nonTCGA_FIN_AC  ExAC_nonTCGA_FIN_AF  ExAC_nonTCGA_NFE_AC  ExAC_nonTCGA_NFE_AF  ExAC_nonTCGA_SAS_AC  ExAC_nonTCGA_SAS_AF  ExAC_nonpsych_AC  ExAC_nonpsych_AF  \\\n",
       "0                       0.0                             0.01567  0.2547:0.0:0.5282:0.2171               3.5592                        0.07372         NaN         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN         NaN         NaN        NaN        NaN            NaN            NaN            NaN            NaN      NaN      NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN              NaN              NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN               NaN               NaN   \n",
       "1                       0.0                             0.01567  0.2547:0.0:0.5282:0.2171               3.5592                        0.07372         NaN         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN         NaN         NaN        NaN        NaN            NaN            NaN            NaN            NaN      NaN      NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN              NaN              NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN               NaN               NaN   \n",
       "\n",
       "   ExAC_nonpsych_Adj_AC  ExAC_nonpsych_Adj_AF  ExAC_nonpsych_AFR_AC  ExAC_nonpsych_AFR_AF  ExAC_nonpsych_AMR_AC  ExAC_nonpsych_AMR_AF  ExAC_nonpsych_EAS_AC  ExAC_nonpsych_EAS_AF  ExAC_nonpsych_FIN_AC  ExAC_nonpsych_FIN_AF  ExAC_nonpsych_NFE_AC  ExAC_nonpsych_NFE_AF  ExAC_nonpsych_SAS_AC  ExAC_nonpsych_SAS_AF  clinvar_rs  clinvar_clnsig  clinvar_trait  clinvar_golden_stars  Interpro_domain  GTEx_V6_gene  GTEx_V6_tissue  \n",
       "0                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN         NaN             NaN            NaN                   NaN              NaN           NaN             NaN  \n",
       "1                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN         NaN             NaN            NaN                   NaN              NaN           NaN             NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "variant_chr1 = pd.read_csv(\n",
    "    op.join(NOTEBOOK_NAME, 'dbNSFP3.2a_variant.chr1'), sep='\\t', nrows=100, na_values=['.', '\\\\N'])\n",
    "display(variant_chr1.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos_1based</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>aaref</th>\n",
       "      <th>aaalt</th>\n",
       "      <th>rs_dbsnp146</th>\n",
       "      <th>hg19_chr</th>\n",
       "      <th>hg19_pos_1based</th>\n",
       "      <th>hg18_chr</th>\n",
       "      <th>hg18_pos_1based</th>\n",
       "      <th>genename</th>\n",
       "      <th>cds_strand</th>\n",
       "      <th>refcodon</th>\n",
       "      <th>codonpos</th>\n",
       "      <th>codon_degeneracy</th>\n",
       "      <th>ancestral_allele</th>\n",
       "      <th>altai_neandertal</th>\n",
       "      <th>denisova</th>\n",
       "      <th>ensembl_geneid</th>\n",
       "      <th>ensembl_transcriptid</th>\n",
       "      <th>ensembl_proteinid</th>\n",
       "      <th>aapos</th>\n",
       "      <th>sift_score</th>\n",
       "      <th>sift_converted_rankscore</th>\n",
       "      <th>sift_pred</th>\n",
       "      <th>uniprot_acc_polyphen2</th>\n",
       "      <th>uniprot_id_polyphen2</th>\n",
       "      <th>uniprot_aapos_polyphen2</th>\n",
       "      <th>polyphen2_hdiv_score</th>\n",
       "      <th>polyphen2_hdiv_rankscore</th>\n",
       "      <th>polyphen2_hdiv_pred</th>\n",
       "      <th>polyphen2_hvar_score</th>\n",
       "      <th>polyphen2_hvar_rankscore</th>\n",
       "      <th>polyphen2_hvar_pred</th>\n",
       "      <th>lrt_score</th>\n",
       "      <th>lrt_converted_rankscore</th>\n",
       "      <th>lrt_pred</th>\n",
       "      <th>lrt_omega</th>\n",
       "      <th>mutationtaster_score</th>\n",
       "      <th>mutationtaster_converted_rankscore</th>\n",
       "      <th>mutationtaster_pred</th>\n",
       "      <th>mutationtaster_model</th>\n",
       "      <th>mutationtaster_aae</th>\n",
       "      <th>mutationassessor_uniprotid</th>\n",
       "      <th>mutationassessor_variant</th>\n",
       "      <th>mutationassessor_score</th>\n",
       "      <th>mutationassessor_score_rankscore</th>\n",
       "      <th>mutationassessor_pred</th>\n",
       "      <th>fathmm_score</th>\n",
       "      <th>fathmm_converted_rankscore</th>\n",
       "      <th>fathmm_pred</th>\n",
       "      <th>provean_score</th>\n",
       "      <th>provean_converted_rankscore</th>\n",
       "      <th>provean_pred</th>\n",
       "      <th>transcript_id_vest3</th>\n",
       "      <th>transcript_var_vest3</th>\n",
       "      <th>vest3_score</th>\n",
       "      <th>vest3_rankscore</th>\n",
       "      <th>metasvm_score</th>\n",
       "      <th>metasvm_rankscore</th>\n",
       "      <th>metasvm_pred</th>\n",
       "      <th>metalr_score</th>\n",
       "      <th>metalr_rankscore</th>\n",
       "      <th>metalr_pred</th>\n",
       "      <th>reliability_index</th>\n",
       "      <th>cadd_raw</th>\n",
       "      <th>cadd_raw_rankscore</th>\n",
       "      <th>cadd_phred</th>\n",
       "      <th>dann_score</th>\n",
       "      <th>dann_rankscore</th>\n",
       "      <th>fathmmmkl_coding_score</th>\n",
       "      <th>fathmmmkl_coding_rankscore</th>\n",
       "      <th>fathmmmkl_coding_pred</th>\n",
       "      <th>fathmmmkl_coding_group</th>\n",
       "      <th>eigenraw</th>\n",
       "      <th>eigenphred</th>\n",
       "      <th>eigenraw_rankscore</th>\n",
       "      <th>eigenpcraw</th>\n",
       "      <th>eigenpcraw_rankscore</th>\n",
       "      <th>genocanyon_score</th>\n",
       "      <th>genocanyon_score_rankscore</th>\n",
       "      <th>integrated_fitcons_score</th>\n",
       "      <th>integrated_fitcons_score_rankscore</th>\n",
       "      <th>integrated_confidence_value</th>\n",
       "      <th>gm12878_fitcons_score</th>\n",
       "      <th>gm12878_fitcons_score_rankscore</th>\n",
       "      <th>gm12878_confidence_value</th>\n",
       "      <th>h1hesc_fitcons_score</th>\n",
       "      <th>h1hesc_fitcons_score_rankscore</th>\n",
       "      <th>h1hesc_confidence_value</th>\n",
       "      <th>huvec_fitcons_score</th>\n",
       "      <th>huvec_fitcons_score_rankscore</th>\n",
       "      <th>huvec_confidence_value</th>\n",
       "      <th>gerp_nr</th>\n",
       "      <th>gerp_rs</th>\n",
       "      <th>gerp_rs_rankscore</th>\n",
       "      <th>phylop100way_vertebrate</th>\n",
       "      <th>phylop100way_vertebrate_rankscore</th>\n",
       "      <th>phylop20way_mammalian</th>\n",
       "      <th>phylop20way_mammalian_rankscore</th>\n",
       "      <th>phastcons100way_vertebrate</th>\n",
       "      <th>phastcons100way_vertebrate_rankscore</th>\n",
       "      <th>phastcons20way_mammalian</th>\n",
       "      <th>phastcons20way_mammalian_rankscore</th>\n",
       "      <th>siphy_29way_pi</th>\n",
       "      <th>siphy_29way_logodds</th>\n",
       "      <th>siphy_29way_logodds_rankscore</th>\n",
       "      <th>gp3_ac</th>\n",
       "      <th>gp3_af</th>\n",
       "      <th>gp3_afr_ac</th>\n",
       "      <th>gp3_afr_af</th>\n",
       "      <th>gp3_eur_ac</th>\n",
       "      <th>gp3_eur_af</th>\n",
       "      <th>gp3_amr_ac</th>\n",
       "      <th>gp3_amr_af</th>\n",
       "      <th>gp3_eas_ac</th>\n",
       "      <th>gp3_eas_af</th>\n",
       "      <th>gp3_sas_ac</th>\n",
       "      <th>gp3_sas_af</th>\n",
       "      <th>twinsuk_ac</th>\n",
       "      <th>twinsuk_af</th>\n",
       "      <th>alspac_ac</th>\n",
       "      <th>alspac_af</th>\n",
       "      <th>esp6500_aa_ac</th>\n",
       "      <th>esp6500_aa_af</th>\n",
       "      <th>esp6500_ea_ac</th>\n",
       "      <th>esp6500_ea_af</th>\n",
       "      <th>exac_ac</th>\n",
       "      <th>exac_af</th>\n",
       "      <th>exac_adj_ac</th>\n",
       "      <th>exac_adj_af</th>\n",
       "      <th>exac_afr_ac</th>\n",
       "      <th>exac_afr_af</th>\n",
       "      <th>exac_amr_ac</th>\n",
       "      <th>exac_amr_af</th>\n",
       "      <th>exac_eas_ac</th>\n",
       "      <th>exac_eas_af</th>\n",
       "      <th>exac_fin_ac</th>\n",
       "      <th>exac_fin_af</th>\n",
       "      <th>exac_nfe_ac</th>\n",
       "      <th>exac_nfe_af</th>\n",
       "      <th>exac_sas_ac</th>\n",
       "      <th>exac_sas_af</th>\n",
       "      <th>exac_nontcga_ac</th>\n",
       "      <th>exac_nontcga_af</th>\n",
       "      <th>exac_nontcga_adj_ac</th>\n",
       "      <th>exac_nontcga_adj_af</th>\n",
       "      <th>exac_nontcga_afr_ac</th>\n",
       "      <th>exac_nontcga_afr_af</th>\n",
       "      <th>exac_nontcga_amr_ac</th>\n",
       "      <th>exac_nontcga_amr_af</th>\n",
       "      <th>exac_nontcga_eas_ac</th>\n",
       "      <th>exac_nontcga_eas_af</th>\n",
       "      <th>exac_nontcga_fin_ac</th>\n",
       "      <th>exac_nontcga_fin_af</th>\n",
       "      <th>exac_nontcga_nfe_ac</th>\n",
       "      <th>exac_nontcga_nfe_af</th>\n",
       "      <th>exac_nontcga_sas_ac</th>\n",
       "      <th>exac_nontcga_sas_af</th>\n",
       "      <th>exac_nonpsych_ac</th>\n",
       "      <th>exac_nonpsych_af</th>\n",
       "      <th>exac_nonpsych_adj_ac</th>\n",
       "      <th>exac_nonpsych_adj_af</th>\n",
       "      <th>exac_nonpsych_afr_ac</th>\n",
       "      <th>exac_nonpsych_afr_af</th>\n",
       "      <th>exac_nonpsych_amr_ac</th>\n",
       "      <th>exac_nonpsych_amr_af</th>\n",
       "      <th>exac_nonpsych_eas_ac</th>\n",
       "      <th>exac_nonpsych_eas_af</th>\n",
       "      <th>exac_nonpsych_fin_ac</th>\n",
       "      <th>exac_nonpsych_fin_af</th>\n",
       "      <th>exac_nonpsych_nfe_ac</th>\n",
       "      <th>exac_nonpsych_nfe_af</th>\n",
       "      <th>exac_nonpsych_sas_ac</th>\n",
       "      <th>exac_nonpsych_sas_af</th>\n",
       "      <th>clinvar_rs</th>\n",
       "      <th>clinvar_clnsig</th>\n",
       "      <th>clinvar_trait</th>\n",
       "      <th>clinvar_golden_stars</th>\n",
       "      <th>interpro_domain</th>\n",
       "      <th>gtex_v6_gene</th>\n",
       "      <th>gtex_v6_tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>M</td>\n",
       "      <td>L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>1</td>\n",
       "      <td>58954</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>ATG</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENST00000335137</td>\n",
       "      <td>ENSP00000334393</td>\n",
       "      <td>1</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.26702</td>\n",
       "      <td>T</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02634</td>\n",
       "      <td>B</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01257</td>\n",
       "      <td>B</td>\n",
       "      <td>0.589091</td>\n",
       "      <td>0.05577</td>\n",
       "      <td>N</td>\n",
       "      <td>1.33974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>M1L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.76</td>\n",
       "      <td>0.00529</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.38</td>\n",
       "      <td>0.13435</td>\n",
       "      <td>N</td>\n",
       "      <td>NM_001005484.1</td>\n",
       "      <td>M1L</td>\n",
       "      <td>0.120</td>\n",
       "      <td>0.13350</td>\n",
       "      <td>-0.9577</td>\n",
       "      <td>0.39629</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>0.00318</td>\n",
       "      <td>T</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-1.436194</td>\n",
       "      <td>0.01840</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.443783</td>\n",
       "      <td>0.03370</td>\n",
       "      <td>0.02063</td>\n",
       "      <td>0.06083</td>\n",
       "      <td>N</td>\n",
       "      <td>AEFI</td>\n",
       "      <td>-1.394131</td>\n",
       "      <td>0.119256</td>\n",
       "      <td>0.02697</td>\n",
       "      <td>-1.535705</td>\n",
       "      <td>0.02036</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.0355</td>\n",
       "      <td>0.487112</td>\n",
       "      <td>0.13308</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.26071</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.22998</td>\n",
       "      <td>0</td>\n",
       "      <td>0.564101</td>\n",
       "      <td>0.26208</td>\n",
       "      <td>0</td>\n",
       "      <td>2.31</td>\n",
       "      <td>-4.63</td>\n",
       "      <td>0.03101</td>\n",
       "      <td>-0.055</td>\n",
       "      <td>0.11668</td>\n",
       "      <td>-1.983</td>\n",
       "      <td>0.00506</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.06329</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01567</td>\n",
       "      <td>0.2547:0.0:0.5282:0.2171</td>\n",
       "      <td>3.5592</td>\n",
       "      <td>0.07372</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>M</td>\n",
       "      <td>V</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>69091</td>\n",
       "      <td>1</td>\n",
       "      <td>58954</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>+</td>\n",
       "      <td>ATG</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>ENSG00000186092</td>\n",
       "      <td>ENST00000335137</td>\n",
       "      <td>ENSP00000334393</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.00963</td>\n",
       "      <td>T</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02634</td>\n",
       "      <td>B</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01257</td>\n",
       "      <td>B</td>\n",
       "      <td>0.589091</td>\n",
       "      <td>0.05577</td>\n",
       "      <td>N</td>\n",
       "      <td>1.33974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OR4F5_HUMAN</td>\n",
       "      <td>M1V</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.76</td>\n",
       "      <td>0.00529</td>\n",
       "      <td>T</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.02137</td>\n",
       "      <td>N</td>\n",
       "      <td>NM_001005484.1</td>\n",
       "      <td>M1V</td>\n",
       "      <td>0.059</td>\n",
       "      <td>0.04258</td>\n",
       "      <td>-0.9620</td>\n",
       "      <td>0.38819</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0005</td>\n",
       "      <td>0.00188</td>\n",
       "      <td>T</td>\n",
       "      <td>8.0</td>\n",
       "      <td>-3.657273</td>\n",
       "      <td>0.00025</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.263843</td>\n",
       "      <td>0.01262</td>\n",
       "      <td>0.00266</td>\n",
       "      <td>0.01460</td>\n",
       "      <td>N</td>\n",
       "      <td>AEFI</td>\n",
       "      <td>-1.588121</td>\n",
       "      <td>0.057701</td>\n",
       "      <td>0.01327</td>\n",
       "      <td>-1.701456</td>\n",
       "      <td>0.01108</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.0355</td>\n",
       "      <td>0.487112</td>\n",
       "      <td>0.13308</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.26071</td>\n",
       "      <td>0</td>\n",
       "      <td>0.573888</td>\n",
       "      <td>0.22998</td>\n",
       "      <td>0</td>\n",
       "      <td>0.564101</td>\n",
       "      <td>0.26208</td>\n",
       "      <td>0</td>\n",
       "      <td>2.31</td>\n",
       "      <td>-4.63</td>\n",
       "      <td>0.03101</td>\n",
       "      <td>-0.055</td>\n",
       "      <td>0.11668</td>\n",
       "      <td>-1.983</td>\n",
       "      <td>0.00506</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.06329</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.01567</td>\n",
       "      <td>0.2547:0.0:0.5282:0.2171</td>\n",
       "      <td>3.5592</td>\n",
       "      <td>0.07372</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr  pos_1based ref alt aaref aaalt rs_dbsnp146  hg19_chr  hg19_pos_1based  hg18_chr  hg18_pos_1based genename cds_strand refcodon  codonpos  codon_degeneracy  ancestral_allele altai_neandertal denisova   ensembl_geneid ensembl_transcriptid ensembl_proteinid  aapos  sift_score  sift_converted_rankscore sift_pred uniprot_acc_polyphen2 uniprot_id_polyphen2  uniprot_aapos_polyphen2  polyphen2_hdiv_score  polyphen2_hdiv_rankscore polyphen2_hdiv_pred  polyphen2_hvar_score  polyphen2_hvar_rankscore polyphen2_hvar_pred  lrt_score  lrt_converted_rankscore lrt_pred  lrt_omega  mutationtaster_score  mutationtaster_converted_rankscore mutationtaster_pred mutationtaster_model mutationtaster_aae mutationassessor_uniprotid mutationassessor_variant  mutationassessor_score  mutationassessor_score_rankscore mutationassessor_pred  fathmm_score  fathmm_converted_rankscore fathmm_pred  provean_score  provean_converted_rankscore provean_pred transcript_id_vest3 transcript_var_vest3  vest3_score  \\\n",
       "0    1       69091   A   C     M     L         NaN         1            69091         1            58954    OR4F5          +      ATG         1                 0               NaN              ./.      ./.  ENSG00000186092      ENST00000335137   ENSP00000334393      1        0.13                   0.26702         T                Q8NH21          OR4F5_HUMAN                      1.0                   0.0                   0.02634                   B                   0.0                   0.01257                   B   0.589091                  0.05577        N    1.33974                   NaN                                 NaN                 NaN                  NaN                NaN                OR4F5_HUMAN                      M1L                     NaN                               NaN                   NaN          6.76                     0.00529           T          -0.38                      0.13435            N      NM_001005484.1                  M1L        0.120   \n",
       "1    1       69091   A   G     M     V         NaN         1            69091         1            58954    OR4F5          +      ATG         1                 0               NaN              ./.      ./.  ENSG00000186092      ENST00000335137   ENSP00000334393      1        1.00                   0.00963         T                Q8NH21          OR4F5_HUMAN                      1.0                   0.0                   0.02634                   B                   0.0                   0.01257                   B   0.589091                  0.05577        N    1.33974                   NaN                                 NaN                 NaN                  NaN                NaN                OR4F5_HUMAN                      M1V                     NaN                               NaN                   NaN          6.76                     0.00529           T           0.71                      0.02137            N      NM_001005484.1                  M1V        0.059   \n",
       "\n",
       "   vest3_rankscore  metasvm_score  metasvm_rankscore metasvm_pred  metalr_score  metalr_rankscore metalr_pred  reliability_index  cadd_raw  cadd_raw_rankscore  cadd_phred  dann_score  dann_rankscore  fathmmmkl_coding_score  fathmmmkl_coding_rankscore fathmmmkl_coding_pred fathmmmkl_coding_group  eigenraw  eigenphred  eigenraw_rankscore  eigenpcraw  eigenpcraw_rankscore  genocanyon_score  genocanyon_score_rankscore  integrated_fitcons_score  integrated_fitcons_score_rankscore  integrated_confidence_value  gm12878_fitcons_score  gm12878_fitcons_score_rankscore  gm12878_confidence_value  h1hesc_fitcons_score  h1hesc_fitcons_score_rankscore  h1hesc_confidence_value  huvec_fitcons_score  huvec_fitcons_score_rankscore  huvec_confidence_value  gerp_nr  gerp_rs  gerp_rs_rankscore  phylop100way_vertebrate  phylop100way_vertebrate_rankscore  phylop20way_mammalian  phylop20way_mammalian_rankscore  phastcons100way_vertebrate  phastcons100way_vertebrate_rankscore  phastcons20way_mammalian  \\\n",
       "0          0.13350        -0.9577            0.39629            T        0.0009           0.00318           T                8.0 -1.436194             0.01840       0.003    0.443783         0.03370                 0.02063                     0.06083                     N                   AEFI -1.394131    0.119256             0.02697   -1.535705               0.02036          0.000022                      0.0355                  0.487112                             0.13308                            0               0.573888                          0.26071                         0              0.573888                         0.22998                        0             0.564101                        0.26208                       0     2.31    -4.63            0.03101                   -0.055                            0.11668                 -1.983                          0.00506                         0.0                               0.06329                       0.0   \n",
       "1          0.04258        -0.9620            0.38819            T        0.0005           0.00188           T                8.0 -3.657273             0.00025       0.001    0.263843         0.01262                 0.00266                     0.01460                     N                   AEFI -1.588121    0.057701             0.01327   -1.701456               0.01108          0.000022                      0.0355                  0.487112                             0.13308                            0               0.573888                          0.26071                         0              0.573888                         0.22998                        0             0.564101                        0.26208                       0     2.31    -4.63            0.03101                   -0.055                            0.11668                 -1.983                          0.00506                         0.0                               0.06329                       0.0   \n",
       "\n",
       "   phastcons20way_mammalian_rankscore            siphy_29way_pi  siphy_29way_logodds  siphy_29way_logodds_rankscore  gp3_ac  gp3_af  gp3_afr_ac  gp3_afr_af  gp3_eur_ac  gp3_eur_af  gp3_amr_ac  gp3_amr_af  gp3_eas_ac  gp3_eas_af  gp3_sas_ac  gp3_sas_af  twinsuk_ac  twinsuk_af  alspac_ac  alspac_af  esp6500_aa_ac  esp6500_aa_af  esp6500_ea_ac  esp6500_ea_af  exac_ac  exac_af  exac_adj_ac  exac_adj_af  exac_afr_ac  exac_afr_af  exac_amr_ac  exac_amr_af  exac_eas_ac  exac_eas_af  exac_fin_ac  exac_fin_af  exac_nfe_ac  exac_nfe_af  exac_sas_ac  exac_sas_af  exac_nontcga_ac  exac_nontcga_af  exac_nontcga_adj_ac  exac_nontcga_adj_af  exac_nontcga_afr_ac  exac_nontcga_afr_af  exac_nontcga_amr_ac  exac_nontcga_amr_af  exac_nontcga_eas_ac  exac_nontcga_eas_af  exac_nontcga_fin_ac  exac_nontcga_fin_af  exac_nontcga_nfe_ac  exac_nontcga_nfe_af  exac_nontcga_sas_ac  exac_nontcga_sas_af  exac_nonpsych_ac  exac_nonpsych_af  exac_nonpsych_adj_ac  exac_nonpsych_adj_af  exac_nonpsych_afr_ac  \\\n",
       "0                             0.01567  0.2547:0.0:0.5282:0.2171               3.5592                        0.07372     NaN     NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN        NaN        NaN            NaN            NaN            NaN            NaN      NaN      NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN              NaN              NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN               NaN               NaN                   NaN                   NaN                   NaN   \n",
       "1                             0.01567  0.2547:0.0:0.5282:0.2171               3.5592                        0.07372     NaN     NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN         NaN        NaN        NaN            NaN            NaN            NaN            NaN      NaN      NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN          NaN              NaN              NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN                  NaN               NaN               NaN                   NaN                   NaN                   NaN   \n",
       "\n",
       "   exac_nonpsych_afr_af  exac_nonpsych_amr_ac  exac_nonpsych_amr_af  exac_nonpsych_eas_ac  exac_nonpsych_eas_af  exac_nonpsych_fin_ac  exac_nonpsych_fin_af  exac_nonpsych_nfe_ac  exac_nonpsych_nfe_af  exac_nonpsych_sas_ac  exac_nonpsych_sas_af  clinvar_rs  clinvar_clnsig  clinvar_trait  clinvar_golden_stars  interpro_domain  gtex_v6_gene  gtex_v6_tissue  \n",
       "0                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN         NaN             NaN            NaN                   NaN              NaN           NaN             NaN  \n",
       "1                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN                   NaN         NaN             NaN            NaN                   NaN              NaN           NaN             NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "variant_chr1.columns = datapkg.format_columns(variant_chr1.columns)\n",
    "display(variant_chr1.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Overwriting load_data/_split_variant_columns.py\n"
     ]
    }
   ],
   "source": [
    "%%file $NOTEBOOK_NAME/_split_variant_columns.py\n",
    "\n",
    "from itertools import chain\n",
    "try:\n",
    "    from itertools import izip_longest as zip_longest\n",
    "except ImportError:\n",
    "    from itertools import zip_longest\n",
    "\n",
    "\n",
    "ensembl_columns = [\n",
    "    (21, 'ensembl_transcriptid'), \n",
    "    (22, 'ensembl_proteinid'), \n",
    "    #\n",
    "    (5, '_aaref'),\n",
    "    (23, 'ensembl_mutation'),\n",
    "    (6, '_aaalt'),\n",
    "    #\n",
    "    (24, 'sift_score'), \n",
    "    (26, 'sift_pred'), \n",
    "    (50, 'fathmm_score'), \n",
    "    (52, 'fathmm_pred'), \n",
    "    (53, 'provean_score'), \n",
    "    (55, 'provean_pred'),\n",
    "]\n",
    "\n",
    "\n",
    "uniprot_columns = [\n",
    "    (27, 'uniprot_acc'), \n",
    "    (28, 'uniprot_id'),\n",
    "    # \n",
    "    (5, '_aaref'),\n",
    "    (29, 'uniprot_mutation'),\n",
    "    (6, '_aaalt'),\n",
    "    #\n",
    "    (30, 'polyphen2_hdiv_score'), \n",
    "    (32, 'polyphen2_hdiv_pred'), \n",
    "    (33, 'polyphen2_hvar_score'), \n",
    "    (35, 'polyphen2_hvar_pred'),\n",
    "]\n",
    "\n",
    "\n",
    "\n",
    "def main(file, start_idx):\n",
    "    \"\"\"Split rows containing comma-separated lists of ensembl ids and uniprot ids into separate files.\"\"\"\n",
    "    ifh = open(file, 'rt')\n",
    "    #\n",
    "    file_ensembl = file + '_ensembl'\n",
    "    file_ensembl_fh = open(file_ensembl, 'w')\n",
    "    file_ensembl_fh.write(\n",
    "        'variant_idx\\t' + '\\t'.join(c[1] for c in ensembl_columns if not c[1].startswith('_')) + '\\n')\n",
    "    #\n",
    "    file_uniprot = file + '_uniprot'\n",
    "    file_uniprot_fh = open(file_uniprot, 'w')\n",
    "    file_uniprot_fh.write(\n",
    "        'variant_idx\\t' + '\\t'.join(c[1] for c in uniprot_columns if not c[1].startswith('_')) + '\\n')\n",
    "    #\n",
    "    next(ifh)  # skip the header\n",
    "    for i, line in enumerate(ifh):\n",
    "        variant_idx = str(start_idx + i)\n",
    "        row = line.strip().split('\\t')\n",
    "        for values in iter_row(row, ensembl_columns, variant_idx):\n",
    "            file_ensembl_fh.write('\\t'.join(values) + '\\n')\n",
    "        for values in iter_row(row, uniprot_columns, variant_idx):\n",
    "            file_uniprot_fh.write('\\t'.join(values) + '\\n')\n",
    "\n",
    "        \n",
    "def iter_row(row, columns, idx):\n",
    "    \"\"\".\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    row : list \n",
    "        List of column values\n",
    "    columns : list of tuples\n",
    "        List of (column_num, column_name) tuples.\n",
    "    idx : int\n",
    "        Becomes the first column.\n",
    "    \"\"\"\n",
    "    # Have to subtract 1 because indexes start from 0\n",
    "    for values in zip_longest(*[row[c[0] - 1].split(',') for c in columns], fillvalue='\\\\N'):\n",
    "        if values[0] == '\\\\N':\n",
    "            continue\n",
    "        mutation = values[2] + values[3] + values[4]\n",
    "        yield chain([idx], values[:2], [mutation], values[5:])\n",
    "        \n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    import sys\n",
    "    import argparse\n",
    "    parser = argparse.ArgumentParser()\n",
    "    parser.add_argument('-f', '--file', type=str)\n",
    "    parser.add_argument('-i', '--idx', type=int)\n",
    "    args = parser.parse_args()\n",
    "    sys.exit(main(args.file, args.idx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def load_table_to_db(suffix, dtypes=None, auto_increment=1):\n",
    "    file = op.abspath(op.join(NOTEBOOK_NAME, 'dbNSFP3.2a_variant.chr{}'.format(suffix)))\n",
    "    tablename = 'variant_chr{}'.format(suffix)\n",
    "    \n",
    "    # Replace ';' to ',' so that FIND_IN_SET works\n",
    "    datapkg.run_command(\"sed -i 's/;/,/g' '{}'\".format(file))\n",
    "    \n",
    "    # Create separate `_ensembl` and `_uniprot` files.\n",
    "    system_command = (\n",
    "        'pypy load_data/_split_variant_columns.py -f {file} -i {auto_increment}'\n",
    "        .format(file=file, auto_increment=auto_increment)\n",
    "    )\n",
    "    p = subprocess.run(\n",
    "        shlex.split(auto_increment), \n",
    "        stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)\n",
    "    try:\n",
    "        p.check_returncode()\n",
    "    except subprocess.CalledProcessError:\n",
    "        print(p.stdout)\n",
    "        print(p.stderr)\n",
    "        raise \n",
    "\n",
    "\n",
    "    # === Main table ===\n",
    "\n",
    "    # Load file to database\n",
    "    t = db.import_table(\n",
    "        filename=file,\n",
    "        tablename=tablename,\n",
    "        # na_values=['\\\\N'],\n",
    "        na_values=['.', '\\\\N'],\n",
    "        dtypes=dtypes,\n",
    "        extra_dtypes={\n",
    "            'chr': sa.VARCHAR(2), \n",
    "            'hg19_chr': sa.VARCHAR(2), 'hg18_chr': sa.VARCHAR(2),\n",
    "            'hg19_pos_1based': sa.INTEGER, 'hg18_pos_1based': sa.INTEGER,\n",
    "        }\n",
    "    )\n",
    "    \n",
    "    # Add an autoincrement 'idx' column\n",
    "    max_id = db.add_id_column(t.name, auto_increment=auto_increment)\n",
    "    \n",
    "    # Create indices\n",
    "    db.create_indexes(\n",
    "        t.name,\n",
    "        index_commands=[\n",
    "            #\n",
    "            [('chr', 'pos_1based', 'ref', 'alt', \n",
    "              'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "            [('hg19_chr', 'hg19_pos_1based', 'ref', 'alt', \n",
    "              'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "            [('hg18_chr', 'hg18_pos_1based', 'ref', 'alt', \n",
    "              'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "            #\n",
    "            [('rs_dbsnp146', 'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "            [('clinvar_rs', 'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "            #\n",
    "            [('genename', 'aaref', 'aaalt', ), False],\n",
    "            [('ensembl_geneid', 'aaref', 'aaalt', ), False],\n",
    "            #\n",
    "            [('mutationassessor_uniprotid ', 'mutationassessor_variant', \n",
    "              'chr', 'pos_1based', 'ref', 'alt', ), False],\n",
    "        ],\n",
    "    )\n",
    "\n",
    "\n",
    "    # === Ensembl table ===\n",
    "\n",
    "    te = db.import_table(\n",
    "        filename=file + '_ensembl',\n",
    "        tablename=tablename + '_ensembl',\n",
    "        na_values=['\\\\N'],\n",
    "    )\n",
    "    db.add_id_column(te.name)\n",
    "    db.create_indexes(\n",
    "        te.name,\n",
    "        index_commands=[\n",
    "            [('ensembl_transcriptid', 'ensembl_mutation', 'fathmm_pred', 'fathmm_score'), True],\n",
    "            [('ensembl_proteinid', 'ensembl_mutation', 'fathmm_pred', 'fathmm_score'), True],\n",
    "        ],\n",
    "    )\n",
    "\n",
    "\n",
    "    # === Uniprot KB table ====\n",
    "\n",
    "    tu = db.import_table(\n",
    "        filename=file + '_uniprot',\n",
    "        tablename=tablename + '_uniprot',\n",
    "        na_values=['\\\\N'],\n",
    "    )\n",
    "    db.add_id_column(tu.name)\n",
    "    db.create_indexes(\n",
    "        tu.name,\n",
    "        index_commands=[\n",
    "            [('uniprot_acc', 'uniprot_mutation', 'polyphen2_hdiv_pred', 'polyphen2_hdiv_score'), True],\n",
    "            [('uniprot_id', 'uniprot_mutation', 'polyphen2_hdiv_pred', 'polyphen2_hdiv_score'), True],\n",
    "        ],\n",
    "    )\n",
    "\n",
    "    return t, max_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG - datapkg._helper - sed -i 's/;/,/g' '/kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant.chr1'\n",
      "DEBUG - datapkg._helper - Running locally\n",
      "DEBUG - datapkg._helper - Command ran successfully:\n",
      "\n",
      "DEBUG - datapkg._helper - File '/kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant' is not compressed...\n",
      "DEBUG - datapkg._helper - sed -i -e 's/\\t\\.\\t/\\t\\\\N\\t/g' -e 's/\\t\\.\\t/\\t\\\\N\\t/g' -e 's/^\\.\\t/\\\\N\\t/g' -e 's/\\t\\.$/\\t\\\\N/g' -e 's/\\t\\.\\r$/\\t\\\\N/g'  /kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant.chr1\n",
      "DEBUG - datapkg._helper - Running locally\n",
      "DEBUG - datapkg._helper - Command ran successfully:\n",
      "\n",
      "DEBUG - datapkg._df_helper - get_file_dtypes(/kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant.chr1, 100000, 100, {'sep': '\\t', 'na_values': ['.', '\\\\N']})\n",
      "DEBUG - datapkg._helper - wc -l '/kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant.chr1'\n",
      "DEBUG - datapkg._helper - Running locally\n",
      "DEBUG - datapkg._helper - Command ran successfully:\n",
      "8500164 /kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant.chr1\n",
      "DEBUG - datapkg._df_helper - Reading file in chunks...\n",
      "DEBUG - datapkg.connections - Loading data into MySQL table: 'variant_chr1'...\n",
      "DEBUG - datapkg._helper - mysql --local-infile --socket=/tmp/strokach/dbnsfp/mysql.sock -u root dbnsfp -e \"load data local infile '/kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant.chr1' into table `variant_chr1` ignore 1 lines;  show warnings;\" \n",
      "DEBUG - datapkg._helper - Running locally\n",
      "DEBUG - datapkg._helper - Command ran successfully:\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Load the first table\n",
    "chr1_t, chr1_max_id = load_table_to_db(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG - datapkg._helper - sed -i 's/;/,/g' '/kimlab1/database_data/biodb/recipes/dbnsfp/notebooks/load_data/dbNSFP3.2a_variant.chr1'\n",
      "DEBUG - datapkg._helper - Running locally\n"
     ]
    }
   ],
   "source": [
    "# Create the remaining tables\n",
    "max_id = chr1_max_id\n",
    "for suffix in list(list(range(2, 23)) + ['M', 'X', 'Y']):\n",
    "    print(suffix)\n",
    "    t, max_id = load_table_to_db(suffix, dtypes=chr1_t.dtypes, auto_increment=max_id+1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['variant_chr1', 'variant_chr2']"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_tables = ['variant_chr{}'.format(i) for i in itertools.chain(range(1, 23), ('M', 'X', 'Y'))]\n",
    "variant_tables[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "variant_chr1\n",
      "variant_chr2\n",
      "variant_chr3\n",
      "variant_chr4\n",
      "variant_chr5\n",
      "variant_chr6\n",
      "variant_chr7\n",
      "variant_chr8\n",
      "variant_chr9\n",
      "variant_chr10\n",
      "variant_chr11\n",
      "variant_chr12\n",
      "variant_chr13\n",
      "variant_chr14\n",
      "variant_chr15\n",
      "variant_chr16\n",
      "variant_chr17\n",
      "variant_chr18\n",
      "variant_chr19\n",
      "variant_chr20\n",
      "variant_chr21\n",
      "variant_chr22\n",
      "variant_chrM\n",
      "variant_chrX\n",
      "variant_chrY\n"
     ]
    }
   ],
   "source": [
    "def create_missing_indexes(table_name):\n",
    "    print(table_name); sys.stdout.flush()\n",
    "    db.create_indexes(\n",
    "        table_name,\n",
    "        index_commands=[\n",
    "            [('chr', 'pos_1based', 'ref', 'alt', 'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "            [('rs_dbsnp146', 'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "            [('clinvar_rs', 'mutationassessor_uniprotid ', 'mutationassessor_variant', ), False],\n",
    "        ],\n",
    "    )\n",
    "\n",
    "import concurrent.futures\n",
    "with concurrent.futures.ThreadPoolExecutor(8) as p:\n",
    "    results = p.map(create_missing_indexes, variant_tables)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pypy: /kimlab1/strokach/anaconda/bin/../lib/./libssl.so.1.0.0: no version information available (required by /kimlab1/strokach/anaconda/bin/../lib/libpypy-c.so)\n",
      "pypy: /kimlab1/strokach/anaconda/bin/../lib/./libssl.so.1.0.0: no version information available (required by /kimlab1/strokach/anaconda/bin/../lib/libpypy-c.so)\n",
      "pypy: /kimlab1/strokach/anaconda/bin/../lib/./libcrypto.so.1.0.0: no version information available (required by /kimlab1/strokach/anaconda/bin/../lib/libpypy-c.so)\n",
      "^CTraceback (most recent call last):\n",
      "  File \"load_data/_split_variant_columns.py\", line 81, in <module>\n",
      "    sys.exit(main(args.file, args.idx))\n",
      "  File \"load_data/_split_variant_columns.py\", line 60, in main\n",
      "    for values in iter_row(row, uniprot_columns, variant_idx):\n",
      "  File \"load_data/_split_variant_columns.py\", line 66, in iter_row\n",
      "    for values in zip_longest(*[row[c[0] - 1].split(',') for c in columns], fillvalue='\\\\N'):\n",
      "KeyboardInterrupt\n",
      "\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CREATE TABLE `variant` (\n",
      "  `id` bigint(20) NOT NULL AUTO_INCREMENT,\n",
      "  `chr` varchar(2) DEFAULT NULL,\n",
      "  `pos_1based` int(11) DEFAULT NULL,\n",
      "  `ref` varchar(32) DEFAULT NULL,\n",
      "  `alt` varchar(32) DEFAULT NULL,\n",
      "  `aaref` varchar(32) DEFAULT NULL,\n",
      "  `aaalt` varchar(32) DEFAULT NULL,\n",
      "  `rs_dbsnp146` varchar(32) DEFAULT NULL,\n",
      "  `hg19_chr` varchar(2) DEFAULT NULL,\n",
      "  `hg19_pos_1based` int(11) DEFAULT NULL,\n",
      "  `hg18_chr` varchar(2) DEFAULT NULL,\n",
      "  `hg18_pos_1based` int(11) DEFAULT NULL,\n",
      "  `genename` varchar(255) DEFAULT NULL,\n",
      "  `cds_strand` varchar(32) DEFAULT NULL,\n",
      "  `refcodon` varchar(255) DEFAULT NULL,\n",
      "  `codonpos` double DEFAULT NULL,\n",
      "  `codon_degeneracy` varchar(255) DEFAULT NULL,\n",
      "  `ancestral_allele` varchar(32) DEFAULT NULL,\n",
      "  `altai_neandertal` varchar(32) DEFAULT NULL,\n",
      "  `denisova` varchar(32) DEFAULT NULL,\n",
      "  `ensembl_geneid` varchar(255) DEFAULT NULL,\n",
      "  `ensembl_transcriptid` mediumtext,\n",
      "  `ensembl_proteinid` mediumtext,\n",
      "  `aapos` varchar(255) DEFAULT NULL,\n",
      "  `sift_score` varchar(255) DEFAULT NULL,\n",
      "  `sift_converted_rankscore` double DEFAULT NULL,\n",
      "  `sift_pred` varchar(255) DEFAULT NULL,\n",
      "  `uniprot_acc_polyphen2` mediumtext,\n",
      "  `uniprot_id_polyphen2` varchar(255) DEFAULT NULL,\n",
      "  `uniprot_aapos_polyphen2` varchar(255) DEFAULT NULL,\n",
      "  `polyphen2_hdiv_score` varchar(255) DEFAULT NULL,\n",
      "  `polyphen2_hdiv_rankscore` double DEFAULT NULL,\n",
      "  `polyphen2_hdiv_pred` varchar(255) DEFAULT NULL,\n",
      "  `polyphen2_hvar_score` varchar(255) DEFAULT NULL,\n",
      "  `polyphen2_hvar_rankscore` double DEFAULT NULL,\n",
      "  `polyphen2_hvar_pred` varchar(255) DEFAULT NULL,\n",
      "  `lrt_score` double DEFAULT NULL,\n",
      "  `lrt_converted_rankscore` double DEFAULT NULL,\n",
      "  `lrt_pred` varchar(32) DEFAULT NULL,\n",
      "  `lrt_omega` double DEFAULT NULL,\n",
      "  `mutationtaster_score` varchar(255) DEFAULT NULL,\n",
      "  `mutationtaster_converted_rankscore` double DEFAULT NULL,\n",
      "  `mutationtaster_pred` varchar(255) DEFAULT NULL,\n",
      "  `mutationtaster_model` mediumtext,\n",
      "  `mutationtaster_aae` mediumtext,\n",
      "  `mutationassessor_uniprotid` varchar(32) DEFAULT NULL,\n",
      "  `mutationassessor_variant` varchar(32) DEFAULT NULL,\n",
      "  `mutationassessor_score` double DEFAULT NULL,\n",
      "  `mutationassessor_score_rankscore` double DEFAULT NULL,\n",
      "  `mutationassessor_pred` varchar(32) DEFAULT NULL,\n",
      "  `fathmm_score` varchar(255) DEFAULT NULL,\n",
      "  `fathmm_converted_rankscore` double DEFAULT NULL,\n",
      "  `fathmm_pred` varchar(255) DEFAULT NULL,\n",
      "  `provean_score` varchar(255) DEFAULT NULL,\n",
      "  `provean_converted_rankscore` double DEFAULT NULL,\n",
      "  `provean_pred` varchar(255) DEFAULT NULL,\n",
      "  `transcript_id_vest3` mediumtext,\n",
      "  `transcript_var_vest3` mediumtext,\n",
      "  `vest3_score` mediumtext,\n",
      "  `vest3_rankscore` double DEFAULT NULL,\n",
      "  `metasvm_score` double DEFAULT NULL,\n",
      "  `metasvm_rankscore` double DEFAULT NULL,\n",
      "  `metasvm_pred` varchar(32) DEFAULT NULL,\n",
      "  `metalr_score` double DEFAULT NULL,\n",
      "  `metalr_rankscore` double DEFAULT NULL,\n",
      "  `metalr_pred` varchar(32) DEFAULT NULL,\n",
      "  `reliability_index` double DEFAULT NULL,\n",
      "  `cadd_raw` double DEFAULT NULL,\n",
      "  `cadd_raw_rankscore` double DEFAULT NULL,\n",
      "  `cadd_phred` double DEFAULT NULL,\n",
      "  `dann_score` double DEFAULT NULL,\n",
      "  `dann_rankscore` double DEFAULT NULL,\n",
      "  `fathmmmkl_coding_score` double DEFAULT NULL,\n",
      "  `fathmmmkl_coding_rankscore` double DEFAULT NULL,\n",
      "  `fathmmmkl_coding_pred` varchar(32) DEFAULT NULL,\n",
      "  `fathmmmkl_coding_group` varchar(32) DEFAULT NULL,\n",
      "  `eigenraw` double DEFAULT NULL,\n",
      "  `eigenphred` double DEFAULT NULL,\n",
      "  `eigenraw_rankscore` double DEFAULT NULL,\n",
      "  `eigenpcraw` double DEFAULT NULL,\n",
      "  `eigenpcraw_rankscore` double DEFAULT NULL,\n",
      "  `genocanyon_score` double DEFAULT NULL,\n",
      "  `genocanyon_score_rankscore` double DEFAULT NULL,\n",
      "  `integrated_fitcons_score` double DEFAULT NULL,\n",
      "  `integrated_fitcons_score_rankscore` double DEFAULT NULL,\n",
      "  `integrated_confidence_value` double DEFAULT NULL,\n",
      "  `gm12878_fitcons_score` double DEFAULT NULL,\n",
      "  `gm12878_fitcons_score_rankscore` double DEFAULT NULL,\n",
      "  `gm12878_confidence_value` double DEFAULT NULL,\n",
      "  `h1hesc_fitcons_score` double DEFAULT NULL,\n",
      "  `h1hesc_fitcons_score_rankscore` double DEFAULT NULL,\n",
      "  `h1hesc_confidence_value` double DEFAULT NULL,\n",
      "  `huvec_fitcons_score` double DEFAULT NULL,\n",
      "  `huvec_fitcons_score_rankscore` double DEFAULT NULL,\n",
      "  `huvec_confidence_value` double DEFAULT NULL,\n",
      "  `gerp_nr` double DEFAULT NULL,\n",
      "  `gerp_rs` double DEFAULT NULL,\n",
      "  `gerp_rs_rankscore` double DEFAULT NULL,\n",
      "  `phylop100way_vertebrate` double DEFAULT NULL,\n",
      "  `phylop100way_vertebrate_rankscore` double DEFAULT NULL,\n",
      "  `phylop20way_mammalian` double DEFAULT NULL,\n",
      "  `phylop20way_mammalian_rankscore` double DEFAULT NULL,\n",
      "  `phastcons100way_vertebrate` double DEFAULT NULL,\n",
      "  `phastcons100way_vertebrate_rankscore` double DEFAULT NULL,\n",
      "  `phastcons20way_mammalian` double DEFAULT NULL,\n",
      "  `phastcons20way_mammalian_rankscore` double DEFAULT NULL,\n",
      "  `siphy_29way_pi` varchar(32) DEFAULT NULL,\n",
      "  `siphy_29way_logodds` double DEFAULT NULL,\n",
      "  `siphy_29way_logodds_rankscore` double DEFAULT NULL,\n",
      "  `gp3_ac` double DEFAULT NULL,\n",
      "  `gp3_af` double DEFAULT NULL,\n",
      "  `gp3_afr_ac` double DEFAULT NULL,\n",
      "  `gp3_afr_af` double DEFAULT NULL,\n",
      "  `gp3_eur_ac` double DEFAULT NULL,\n",
      "  `gp3_eur_af` double DEFAULT NULL,\n",
      "  `gp3_amr_ac` double DEFAULT NULL,\n",
      "  `gp3_amr_af` double DEFAULT NULL,\n",
      "  `gp3_eas_ac` double DEFAULT NULL,\n",
      "  `gp3_eas_af` double DEFAULT NULL,\n",
      "  `gp3_sas_ac` double DEFAULT NULL,\n",
      "  `gp3_sas_af` double DEFAULT NULL,\n",
      "  `twinsuk_ac` double DEFAULT NULL,\n",
      "  `twinsuk_af` double DEFAULT NULL,\n",
      "  `alspac_ac` double DEFAULT NULL,\n",
      "  `alspac_af` double DEFAULT NULL,\n",
      "  `esp6500_aa_ac` double DEFAULT NULL,\n",
      "  `esp6500_aa_af` double DEFAULT NULL,\n",
      "  `esp6500_ea_ac` double DEFAULT NULL,\n",
      "  `esp6500_ea_af` double DEFAULT NULL,\n",
      "  `exac_ac` double DEFAULT NULL,\n",
      "  `exac_af` double DEFAULT NULL,\n",
      "  `exac_adj_ac` double DEFAULT NULL,\n",
      "  `exac_adj_af` double DEFAULT NULL,\n",
      "  `exac_afr_ac` double DEFAULT NULL,\n",
      "  `exac_afr_af` double DEFAULT NULL,\n",
      "  `exac_amr_ac` double DEFAULT NULL,\n",
      "  `exac_amr_af` double DEFAULT NULL,\n",
      "  `exac_eas_ac` double DEFAULT NULL,\n",
      "  `exac_eas_af` double DEFAULT NULL,\n",
      "  `exac_fin_ac` double DEFAULT NULL,\n",
      "  `exac_fin_af` double DEFAULT NULL,\n",
      "  `exac_nfe_ac` double DEFAULT NULL,\n",
      "  `exac_nfe_af` double DEFAULT NULL,\n",
      "  `exac_sas_ac` double DEFAULT NULL,\n",
      "  `exac_sas_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_adj_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_adj_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_afr_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_afr_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_amr_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_amr_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_eas_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_eas_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_fin_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_fin_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_nfe_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_nfe_af` double DEFAULT NULL,\n",
      "  `exac_nontcga_sas_ac` double DEFAULT NULL,\n",
      "  `exac_nontcga_sas_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_adj_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_adj_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_afr_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_afr_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_amr_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_amr_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_eas_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_eas_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_fin_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_fin_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_nfe_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_nfe_af` double DEFAULT NULL,\n",
      "  `exac_nonpsych_sas_ac` double DEFAULT NULL,\n",
      "  `exac_nonpsych_sas_af` double DEFAULT NULL,\n",
      "  `clinvar_rs` varchar(32) DEFAULT NULL,\n",
      "  `clinvar_clnsig` varchar(32) DEFAULT NULL,\n",
      "  `clinvar_trait` mediumtext,\n",
      "  `clinvar_golden_stars` varchar(32) DEFAULT NULL,\n",
      "  `interpro_domain` mediumtext,\n",
      "  `gtex_v6_gene` mediumtext,\n",
      "  `gtex_v6_tissue` mediumtext,\n",
      "  PRIMARY KEY (`id`),\n",
      "  KEY `a` (`chr`,`pos_1based`,`ref`,`alt`),\n",
      "  KEY `b` (`chr`,`pos_1based`,`aaref`,`aaalt`),\n",
      "  KEY `c` (`rs_dbsnp146`),\n",
      "  KEY `d` (`aaref`,`aaalt`),\n",
      "  KEY `e` (`ensembl_geneid`,`aaref`,`aaalt`),\n",
      "  KEY `f` (`genename`),\n",
      "  KEY `g` (`uniprot_id_polyphen2`,`uniprot_aapos_polyphen2`),\n",
      "  KEY `h` (`mutationassessor_uniprotid`,`mutationassessor_variant`),\n",
      "  KEY `i` (`hg19_chr`,`hg19_pos_1based`,`ref`,`alt`),\n",
      "  KEY `j` (`hg18_chr`,`hg18_pos_1based`,`ref`,`alt`),\n",
      "  KEY `k` (`chr`,`pos_1based`,`ref`,`alt`,`mutationassessor_uniprotid`,`mutationassessor_variant`),\n",
      "  KEY `l` (`rs_dbsnp146`,`mutationassessor_uniprotid`,`mutationassessor_variant`),\n",
      "  KEY `m` (`clinvar_rs`,`mutationassessor_uniprotid`,`mutationassessor_variant`)\n",
      ") ENGINE=MERGE \n",
      "  UNION=(variant_chr1, variant_chr2, variant_chr3, variant_chr4, variant_chr5, variant_chr6, variant_chr7, \n",
      "         variant_chr8, variant_chr9, variant_chr10, variant_chr11, variant_chr12, variant_chr13, variant_chr14, \n",
      "         variant_chr15, variant_chr16, variant_chr17, variant_chr18, variant_chr19, variant_chr20, variant_chr21, \n",
      "         variant_chr22, variant_chrM, variant_chrX, variant_chrY)\n",
      "  DEFAULT CHARSET=latin1;\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Create a MERGE table\n",
    "sql_command = \"\"\"\\\n",
    "CREATE TABLE `variant` (\n",
    "  `id` bigint(20) NOT NULL AUTO_INCREMENT,\n",
    "  `chr` varchar(2) DEFAULT NULL,\n",
    "  `pos_1based` int(11) DEFAULT NULL,\n",
    "  `ref` varchar(32) DEFAULT NULL,\n",
    "  `alt` varchar(32) DEFAULT NULL,\n",
    "  `aaref` varchar(32) DEFAULT NULL,\n",
    "  `aaalt` varchar(32) DEFAULT NULL,\n",
    "  `rs_dbsnp146` varchar(32) DEFAULT NULL,\n",
    "  `hg19_chr` varchar(2) DEFAULT NULL,\n",
    "  `hg19_pos_1based` int(11) DEFAULT NULL,\n",
    "  `hg18_chr` varchar(2) DEFAULT NULL,\n",
    "  `hg18_pos_1based` int(11) DEFAULT NULL,\n",
    "  `genename` varchar(255) DEFAULT NULL,\n",
    "  `cds_strand` varchar(32) DEFAULT NULL,\n",
    "  `refcodon` varchar(255) DEFAULT NULL,\n",
    "  `codonpos` double DEFAULT NULL,\n",
    "  `codon_degeneracy` varchar(255) DEFAULT NULL,\n",
    "  `ancestral_allele` varchar(32) DEFAULT NULL,\n",
    "  `altai_neandertal` varchar(32) DEFAULT NULL,\n",
    "  `denisova` varchar(32) DEFAULT NULL,\n",
    "  `ensembl_geneid` varchar(255) DEFAULT NULL,\n",
    "  `ensembl_transcriptid` mediumtext,\n",
    "  `ensembl_proteinid` mediumtext,\n",
    "  `aapos` varchar(255) DEFAULT NULL,\n",
    "  `sift_score` varchar(255) DEFAULT NULL,\n",
    "  `sift_converted_rankscore` double DEFAULT NULL,\n",
    "  `sift_pred` varchar(255) DEFAULT NULL,\n",
    "  `uniprot_acc_polyphen2` mediumtext,\n",
    "  `uniprot_id_polyphen2` varchar(255) DEFAULT NULL,\n",
    "  `uniprot_aapos_polyphen2` varchar(255) DEFAULT NULL,\n",
    "  `polyphen2_hdiv_score` varchar(255) DEFAULT NULL,\n",
    "  `polyphen2_hdiv_rankscore` double DEFAULT NULL,\n",
    "  `polyphen2_hdiv_pred` varchar(255) DEFAULT NULL,\n",
    "  `polyphen2_hvar_score` varchar(255) DEFAULT NULL,\n",
    "  `polyphen2_hvar_rankscore` double DEFAULT NULL,\n",
    "  `polyphen2_hvar_pred` varchar(255) DEFAULT NULL,\n",
    "  `lrt_score` double DEFAULT NULL,\n",
    "  `lrt_converted_rankscore` double DEFAULT NULL,\n",
    "  `lrt_pred` varchar(32) DEFAULT NULL,\n",
    "  `lrt_omega` double DEFAULT NULL,\n",
    "  `mutationtaster_score` varchar(255) DEFAULT NULL,\n",
    "  `mutationtaster_converted_rankscore` double DEFAULT NULL,\n",
    "  `mutationtaster_pred` varchar(255) DEFAULT NULL,\n",
    "  `mutationtaster_model` mediumtext,\n",
    "  `mutationtaster_aae` mediumtext,\n",
    "  `mutationassessor_uniprotid` varchar(32) DEFAULT NULL,\n",
    "  `mutationassessor_variant` varchar(32) DEFAULT NULL,\n",
    "  `mutationassessor_score` double DEFAULT NULL,\n",
    "  `mutationassessor_score_rankscore` double DEFAULT NULL,\n",
    "  `mutationassessor_pred` varchar(32) DEFAULT NULL,\n",
    "  `fathmm_score` varchar(255) DEFAULT NULL,\n",
    "  `fathmm_converted_rankscore` double DEFAULT NULL,\n",
    "  `fathmm_pred` varchar(255) DEFAULT NULL,\n",
    "  `provean_score` varchar(255) DEFAULT NULL,\n",
    "  `provean_converted_rankscore` double DEFAULT NULL,\n",
    "  `provean_pred` varchar(255) DEFAULT NULL,\n",
    "  `transcript_id_vest3` mediumtext,\n",
    "  `transcript_var_vest3` mediumtext,\n",
    "  `vest3_score` mediumtext,\n",
    "  `vest3_rankscore` double DEFAULT NULL,\n",
    "  `metasvm_score` double DEFAULT NULL,\n",
    "  `metasvm_rankscore` double DEFAULT NULL,\n",
    "  `metasvm_pred` varchar(32) DEFAULT NULL,\n",
    "  `metalr_score` double DEFAULT NULL,\n",
    "  `metalr_rankscore` double DEFAULT NULL,\n",
    "  `metalr_pred` varchar(32) DEFAULT NULL,\n",
    "  `reliability_index` double DEFAULT NULL,\n",
    "  `cadd_raw` double DEFAULT NULL,\n",
    "  `cadd_raw_rankscore` double DEFAULT NULL,\n",
    "  `cadd_phred` double DEFAULT NULL,\n",
    "  `dann_score` double DEFAULT NULL,\n",
    "  `dann_rankscore` double DEFAULT NULL,\n",
    "  `fathmmmkl_coding_score` double DEFAULT NULL,\n",
    "  `fathmmmkl_coding_rankscore` double DEFAULT NULL,\n",
    "  `fathmmmkl_coding_pred` varchar(32) DEFAULT NULL,\n",
    "  `fathmmmkl_coding_group` varchar(32) DEFAULT NULL,\n",
    "  `eigenraw` double DEFAULT NULL,\n",
    "  `eigenphred` double DEFAULT NULL,\n",
    "  `eigenraw_rankscore` double DEFAULT NULL,\n",
    "  `eigenpcraw` double DEFAULT NULL,\n",
    "  `eigenpcraw_rankscore` double DEFAULT NULL,\n",
    "  `genocanyon_score` double DEFAULT NULL,\n",
    "  `genocanyon_score_rankscore` double DEFAULT NULL,\n",
    "  `integrated_fitcons_score` double DEFAULT NULL,\n",
    "  `integrated_fitcons_score_rankscore` double DEFAULT NULL,\n",
    "  `integrated_confidence_value` double DEFAULT NULL,\n",
    "  `gm12878_fitcons_score` double DEFAULT NULL,\n",
    "  `gm12878_fitcons_score_rankscore` double DEFAULT NULL,\n",
    "  `gm12878_confidence_value` double DEFAULT NULL,\n",
    "  `h1hesc_fitcons_score` double DEFAULT NULL,\n",
    "  `h1hesc_fitcons_score_rankscore` double DEFAULT NULL,\n",
    "  `h1hesc_confidence_value` double DEFAULT NULL,\n",
    "  `huvec_fitcons_score` double DEFAULT NULL,\n",
    "  `huvec_fitcons_score_rankscore` double DEFAULT NULL,\n",
    "  `huvec_confidence_value` double DEFAULT NULL,\n",
    "  `gerp_nr` double DEFAULT NULL,\n",
    "  `gerp_rs` double DEFAULT NULL,\n",
    "  `gerp_rs_rankscore` double DEFAULT NULL,\n",
    "  `phylop100way_vertebrate` double DEFAULT NULL,\n",
    "  `phylop100way_vertebrate_rankscore` double DEFAULT NULL,\n",
    "  `phylop20way_mammalian` double DEFAULT NULL,\n",
    "  `phylop20way_mammalian_rankscore` double DEFAULT NULL,\n",
    "  `phastcons100way_vertebrate` double DEFAULT NULL,\n",
    "  `phastcons100way_vertebrate_rankscore` double DEFAULT NULL,\n",
    "  `phastcons20way_mammalian` double DEFAULT NULL,\n",
    "  `phastcons20way_mammalian_rankscore` double DEFAULT NULL,\n",
    "  `siphy_29way_pi` varchar(32) DEFAULT NULL,\n",
    "  `siphy_29way_logodds` double DEFAULT NULL,\n",
    "  `siphy_29way_logodds_rankscore` double DEFAULT NULL,\n",
    "  `gp3_ac` double DEFAULT NULL,\n",
    "  `gp3_af` double DEFAULT NULL,\n",
    "  `gp3_afr_ac` double DEFAULT NULL,\n",
    "  `gp3_afr_af` double DEFAULT NULL,\n",
    "  `gp3_eur_ac` double DEFAULT NULL,\n",
    "  `gp3_eur_af` double DEFAULT NULL,\n",
    "  `gp3_amr_ac` double DEFAULT NULL,\n",
    "  `gp3_amr_af` double DEFAULT NULL,\n",
    "  `gp3_eas_ac` double DEFAULT NULL,\n",
    "  `gp3_eas_af` double DEFAULT NULL,\n",
    "  `gp3_sas_ac` double DEFAULT NULL,\n",
    "  `gp3_sas_af` double DEFAULT NULL,\n",
    "  `twinsuk_ac` double DEFAULT NULL,\n",
    "  `twinsuk_af` double DEFAULT NULL,\n",
    "  `alspac_ac` double DEFAULT NULL,\n",
    "  `alspac_af` double DEFAULT NULL,\n",
    "  `esp6500_aa_ac` double DEFAULT NULL,\n",
    "  `esp6500_aa_af` double DEFAULT NULL,\n",
    "  `esp6500_ea_ac` double DEFAULT NULL,\n",
    "  `esp6500_ea_af` double DEFAULT NULL,\n",
    "  `exac_ac` double DEFAULT NULL,\n",
    "  `exac_af` double DEFAULT NULL,\n",
    "  `exac_adj_ac` double DEFAULT NULL,\n",
    "  `exac_adj_af` double DEFAULT NULL,\n",
    "  `exac_afr_ac` double DEFAULT NULL,\n",
    "  `exac_afr_af` double DEFAULT NULL,\n",
    "  `exac_amr_ac` double DEFAULT NULL,\n",
    "  `exac_amr_af` double DEFAULT NULL,\n",
    "  `exac_eas_ac` double DEFAULT NULL,\n",
    "  `exac_eas_af` double DEFAULT NULL,\n",
    "  `exac_fin_ac` double DEFAULT NULL,\n",
    "  `exac_fin_af` double DEFAULT NULL,\n",
    "  `exac_nfe_ac` double DEFAULT NULL,\n",
    "  `exac_nfe_af` double DEFAULT NULL,\n",
    "  `exac_sas_ac` double DEFAULT NULL,\n",
    "  `exac_sas_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_adj_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_adj_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_afr_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_afr_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_amr_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_amr_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_eas_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_eas_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_fin_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_fin_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_nfe_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_nfe_af` double DEFAULT NULL,\n",
    "  `exac_nontcga_sas_ac` double DEFAULT NULL,\n",
    "  `exac_nontcga_sas_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_adj_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_adj_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_afr_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_afr_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_amr_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_amr_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_eas_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_eas_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_fin_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_fin_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_nfe_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_nfe_af` double DEFAULT NULL,\n",
    "  `exac_nonpsych_sas_ac` double DEFAULT NULL,\n",
    "  `exac_nonpsych_sas_af` double DEFAULT NULL,\n",
    "  `clinvar_rs` varchar(32) DEFAULT NULL,\n",
    "  `clinvar_clnsig` varchar(32) DEFAULT NULL,\n",
    "  `clinvar_trait` mediumtext,\n",
    "  `clinvar_golden_stars` varchar(32) DEFAULT NULL,\n",
    "  `interpro_domain` mediumtext,\n",
    "  `gtex_v6_gene` mediumtext,\n",
    "  `gtex_v6_tissue` mediumtext,\n",
    "  PRIMARY KEY (`id`),\n",
    "  KEY `a` (`chr`,`pos_1based`,`ref`,`alt`),\n",
    "  KEY `b` (`chr`,`pos_1based`,`aaref`,`aaalt`),\n",
    "  KEY `c` (`rs_dbsnp146`),\n",
    "  KEY `d` (`aaref`,`aaalt`),\n",
    "  KEY `e` (`ensembl_geneid`,`aaref`,`aaalt`),\n",
    "  KEY `f` (`genename`),\n",
    "  KEY `g` (`uniprot_id_polyphen2`,`uniprot_aapos_polyphen2`),\n",
    "  KEY `h` (`mutationassessor_uniprotid`,`mutationassessor_variant`),\n",
    "  KEY `i` (`hg19_chr`,`hg19_pos_1based`,`ref`,`alt`),\n",
    "  KEY `j` (`hg18_chr`,`hg18_pos_1based`,`ref`,`alt`),\n",
    "  KEY `k` (`chr`,`pos_1based`,`ref`,`alt`,`mutationassessor_uniprotid`,`mutationassessor_variant`),\n",
    "  KEY `l` (`rs_dbsnp146`,`mutationassessor_uniprotid`,`mutationassessor_variant`),\n",
    "  KEY `m` (`clinvar_rs`,`mutationassessor_uniprotid`,`mutationassessor_variant`)\n",
    ") ENGINE=MERGE \n",
    "  UNION=(variant_chr1, variant_chr2, variant_chr3, variant_chr4, variant_chr5, variant_chr6, variant_chr7, \n",
    "         variant_chr8, variant_chr9, variant_chr10, variant_chr11, variant_chr12, variant_chr13, variant_chr14, \n",
    "         variant_chr15, variant_chr16, variant_chr17, variant_chr18, variant_chr19, variant_chr20, variant_chr21, \n",
    "         variant_chr22, variant_chrM, variant_chrX, variant_chrY)\n",
    "  DEFAULT CHARSET=latin1;\n",
    "\"\"\"\n",
    "print(sql_command)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlalchemy.engine.result.ResultProxy at 0x7f72147cb9b0>"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "db.engine.execute(sql_command)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Compress"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'MySQLDaemon' object has no attribute 'data_dir'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-38-be7d2d9e0d27>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mmysqld\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmyisampack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mSCHEMA_NAME\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m/home/kimlab1/strokach/working/datapkg/datapkg/daemons.py\u001b[0m in \u001b[0;36mmyisampack\u001b[1;34m(self, schema_name)\u001b[0m\n\u001b[0;32m     61\u001b[0m         data_files = [\n\u001b[0;32m     62\u001b[0m             \u001b[0mop\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mabspath\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mop\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdata_dir\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mschema_name\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 63\u001b[1;33m             \u001b[1;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mlistdir\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mop\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdata_dir\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mschema_name\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     64\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mop\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplitext\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mf\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m-\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'.MYI'\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     65\u001b[0m         ]\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'MySQLDaemon' object has no attribute 'data_dir'"
     ]
    }
   ],
   "source": [
    "mysqld.myisampack(SCHEMA_NAME)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
